ERCS/erc-7092.md

```diff
@@ -1,326 +1,156 @@
-    */
     function principalOf(address _account) external view returns(uint256);
+    function allowance(address _owner, address _spender) external view returns(uint256);
 
-    /**
-    * @notice Returns the amount of tokens the `_spender` account has been authorized by the `_owner``
-    *         acount to manage their bonds
-    * @param _owner the bondholder address
-    * @param _spender the address that has been authorized by the bondholder
-    */
-    function approval(address _owner, address _spender) external view returns(uint256);
-
-    /**
-    * @notice Authorizes `_spender` account to manage `_amount`of their bonds
-    * @param _spender the address to be authorized by the bondholder
-    * @param _amount amount of bond to approve. _amount MUST be a multiple of denomination
-    */
+    // setter functions
     function approve(address _spender, uint256 _amount) external returns(bool);
+    function decreaseAllowance(address _spender, uint256 _amount) external;
+    function transfer(address _to, uint256 _amount, bytes calldata _data) external returns(bool);
+    function transferFrom(address _from, address _to, uint256 _amount, bytes calldata _data) external returns(bool);
 
-    /**
-    * @notice Authorizes the `_spender` account to manage all their bonds
-    * @param _spender the address to be authorized by the bondholder
-    *
-    * OPTIONAL - interfaces and other contracts MUST NOT expect these values to be present. The method is used to improve usability.
-    */
-    function approveAll(address _spender) external returns(bool);
+    // batch functions
+    function batchApprove(address[] calldata _spender, uint256[] calldata _amount) external returns(bool);
+    function batchDecreaseAllowance(address[] calldata _spender, uint256[] calldata _amount) external;
+    function batchTransfer(address[] calldata _to, uint256[] calldata _amount, bytes[] calldata _data) external returns(bool);
+    function batchTransferFrom(address[] calldata _from, address[] calldata _to, uint256[] calldata _amount, bytes[] calldata _data) external returns(bool);
+}
+```
 
-    /**
-    * @notice Lower the allowance of `_spender` by `_amount`
-    * @param _spender the address to be authorized by the bondholder
-    * @param _amount amount of bond to remove approval; _amount MUST be a multiple of denomination
-    */
-    function decreaseAllowance(address _spender, uint256 _amount) external returns(bool);
+### Additional bond parameters Interface
 
-    /**
-    * @notice Remove the allowance for `_spender`
-    * @param _spender the address to remove the authorization by from
-    *
-    * OPTIONAL - interfaces and other contracts MUST NOT expect these values to be present. The method is used to improve usability.
-    */
-    function decreaseAllowanceForAll(address _spender) external returns(bool);
+Functions in the `IERC7092ESG` interface are OPTIONAL. This interface MAY be used to improve the standard usability.
```

Comment:
This is a tad ambiguous. Is the whole interface optional, or are the individual functions within it optional? The former is fine, but the latter will mess up the ERC-165 interface id.

If the whole interface is optional, I'd recommend:

```suggestion
The `IERC7092ESG` interface is OPTIONAL for contracts implementing this proposal. This interface MAY be used to improve the standard usability.
```

On the other hand, if the individual functions are optional, you could move each one into its own interface.